builtin/providers/cloudstack/resource_cloudstack_network.go

```diff
@@ -166,6 +196,8 @@ func resourceCloudStackNetworkRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("name", n.Name)
 	d.Set("display_text", n.Displaytext)
 	d.Set("cidr", n.Cidr)
+	d.Set("gateway", n.Gateway)
+	d.Set("vlan", n.Vlan)
```

Comment:
I noticed a small thing about the `vlan` param in docs:

```
vlan     The vlan of the network. This parameter is visible to ROOT admins only
```

The problem here is that I don't expect people to use (or even have) a ROOT admin account for deploying infra. So for people deploying with a normal admin account, this value will always be the zero value of this field being an empty string. And that would in turn trigger a diff between your config and your state which will never get in sync.

So my question is if you can add a `vlan` to the network create call as a normal admin? If that is the case, I think we can keep the `vlan` param, but we should not try to read it back here...

Does that make sense?